File: Python/백준/Bronze/14626. ISBN/ISBN.py
S = input()
n = 0
idx = 0
for i in range(13):
    if S[i] == '*':
        idx = i
        continue
    if i%2 == 0:
        n += int(S[i])
    else:
        n += 3*int(S[i])

x = 10-(n%10)
if idx%2 == 0:
    print(x%10)
else:
    while x//3 != x/3:
        x += 10
    print((x//3)%10)

File: Java/백준/Bronze/14626. ISBN/ISBN.java
    import java.io.IOException;
    import java.util.*;

    class Main {
        public static void main(String[] args) throws IOException {
            Scanner sc = new Scanner(System.in);
            String S = sc.next();
            int sum = 0;
            int idx = 0;
            for (int i=0; i<S.length(); i++){
                char c = S.charAt(i);
                if (c=='*'){
                    idx = i;
                    continue;
                }
                int n = Integer.valueOf(c) - '0';
                sum += n;
                if (i%2==1){
                    sum += 2*n;
                }
            }
            if (idx % 2 == 0) { 
            for (int x = 0; x <= 9; x++) {
                if ((sum + x) % 10 == 0) {
                    System.out.println(x);
                    break;
                }
            }
        } else {
            for (int x = 0; x <= 9; x++) {
                if ((sum + 3 * x) % 10 == 0) {
                    System.out.println(x);
                    break;
                }
            }
        }
    }
}